packages/front-end/components/Owner/SelectOwner.tsx

```diff
@@ -0,0 +1,79 @@
+import { Text } from "@radix-ui/themes";
+import metaDataStyles from "@/components/Radix/Styles/Metadata.module.scss";
+import useMembers from "@/hooks/useMembers";
+import UserAvatar from "../Avatar/UserAvatar";
+import SelectField from "../Forms/SelectField";
+
+interface Props {
+  value: string;
+  onChange: (v: string) => void;
+  resourceType:
+    | "dimension"
+    | "feature"
+    | "experiment"
+    | "segment"
+    | "factSegment"
+    | "savedGroup"
+    | "metric"
+    | "factMetric"
+    | "archetype"
+    | "factTable";
+  placeholder?: string;
+}
+
+export default function SelectOwner({
+  value,
+  onChange,
+  placeholder = "",
+  resourceType,
+}: Props) {
+  const { memberUsernameOptions, memberUserNameAndIdOptions } = useMembers();
+
+  const resourcesWithOwnerAsIdField = [
```

Comment:
This name isn't clear - it's inverted. It should be `resourcesWithIdAsOwnerField` or something similar. Probably doesn't even make sense to build this as a named variable. Just do it in line below.

E.G.

```ts
// Some resources store the owner by name and some by id, so check which one it is
  const ownerIdentifierType = ["experiment", "experimentTemplate", "factTable", "archetype"].includes(resourceType)
    ? "id"
    : "name";
```